### PR TITLE
Propose security page and policy for Team-Mail

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[](https://james-jenkins.open-paas.org/blue/organizations/jenkins/Tmail%20build/detail/PR-361/1/pipeline#step-29-log-1804)[](https://james-jenkins.open-paas.org/blue/organizations/jenkins/Tmail%20build/detail/PR-361/1/pipeline#step-29-log-1805)Caused by: org.apache.james.task.UnconfigurableHostnameException: Hostname can not be retrieved

Caused by: java.net.UnknownHostException: ci-james-04: ci-james-04: Name or service not known

Caused by: java.net.UnknownHostException: ci-james-04: Name or service not known
```

lol

**Comment by digitalfredy** [comment]
Hi!

I'm a bit lost about the context this intend to be the most general document about Security? because the referenced one for operators have a clear target but I'm not sure about the public target for this one.

File: security.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Security policy
```
Comment (digitalfredy) [inline security.md:1]: As mentioned before I think we can improve the policy but for sure starting by having this documents is a clear progress and better that do not have no policy at all.

File: security.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+We disclose such security issues only once a released version addressing the issue is available.
+
+# Reporting security vulnerabilities
+
+To ensure safety of our users, security process needs to happen privately.
+
+Here are the steps:
+
+ - 1. Reporter email the issues privately to `openpaas-james[AT]linagora.com`.
```
Comment (digitalfredy) [inline security.md:14]: Duplicated from security.adoc

**Review by digitalfredy — COMMENTED** [review]
Great work!
All my comments are not blockers but will be great if we take some time in the future to improve this documents, in special the structure.

File: docs/modules/ROOT/pages/security.adoc [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+Additional suggestions welcomed.
+
+== Reporting security vulnerabilities
+
+To ensure safety of our users, security process needs to happen privately.
+
+Here are the steps:
+
+- 1. Reporter email the issues privately to `openpaas-james[AT]linagora.com`.
+- 2. We will then evaluate the validity of your report, and write back to you.
```
[the remark this answers is not in the document]
Comment (chibenwa) [reply docs/modules/ROOT/pages/security.adoc:64]: I did took inspiration from ASF that I know well. Volunteer based so they avoid cpmmitment.

 Two weeks sounds reasonable to me for a time to inotial response, as a team, even in summer.

File: security.md (hunk 2 above)
Comment (chibenwa) [reply security.md:14]: This was intendeed. The afoc antora file is "the souce of truth" but I'm not expecting people to find it. Hence I wanted to have a link and critical disclosure information there too, at the root of the repo.

File: docs/modules/ROOT/pages/security.adoc [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+While fuzzing with Jazzer the IMAP parsing stack we discover that crafted APPEND and STATUS IMAP command
+could be used to trigger infinite loops resulting in expensive CPU computations and OutOfMemory exceptions.
+This can be used for a Denial Of Service attack. The IMAP user needs to be authenticated to exploit this vulnerability.
+
+This affected Team-Mail prior to 0.4.1.
+
+*Mitigation*: Upgrade to Team-Mail server version 0.4.1.
+
+== Active steps we are taking toward improving security
```
[the remark this answers is not in the document]
Comment (chibenwa) [reply docs/modules/ROOT/pages/security.adoc:47]: I'd like to sau 'schedule a meeting' bit timezones might be an issie. Happy to get feedback - suggestions.

File: docs/modules/ROOT/pages/security.adoc [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+== Security features and recommendations
+
+Team-Mail allow one user to encrypt his email at REST with PGP keys. xref:tmail-backend/features/encrypted-mailbox.adoc[Read more...]
+
+link:https://github.com/apache/james-project/blob/master/server/apps/distributed-app/docs/modules/ROOT/pages/operate/security.adoc#threats[Threat model] described by the James community fully applies.
+
+link:https://github.com/apache/james-project/blob/master/server/apps/distributed-app/docs/modules/ROOT/pages/operate/security.adoc#best-practices[Best practices] described by the James community fully applies.
+
+== Known security vulnerabilities
```
[the remark this answers is not in the document]
Comment (chibenwa) [reply docs/modules/ROOT/pages/security.adoc:12]: Asf james != team mail . If there's something in james it do not necessarily impact tmail and vice versa.

Here the focus is tmail.

**Review by Arsnael — COMMENTED** [review]
Read it nothing to add

File: security.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+ - 2. We will then evaluate the validity of your report, and write back to you within two weeks. This response time 
+ accounts for vacation and will generally be quicker.
+ - 3. We will propose a fix that we will review with you. 
+ - 4. We will propose a draft for the announcement that we will review with you. 
+ - 4. We will propose you a schedule for the release and the announcements.
```
Comment (vttranlina) [inline security.md:19]:
```suggestion
 - 5. We will propose you a schedule for the release and the announcements.
```

**Review by vttranlina — APPROVED** [review]

**Pull request merged**